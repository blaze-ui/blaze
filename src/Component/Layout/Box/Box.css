.Box-primary{
    width: 300px;
  border: 15px solid green;
  padding: 50px;
  margin: 20px;
}

.Box-secondary{
  width: 300px;
  border: 15px solid red;
  /* background-color: yellow;
  color: black; */
  padding: 50px;
  margin: 20px;
}