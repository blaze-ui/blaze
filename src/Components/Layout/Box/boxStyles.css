.boxPrimary {
  background-color: #be1e6b;
  color: #000000;
}

.boxSecondary {
  background-color: #692478;
  color: #000000;
}

.boxTertiary {
  background-color: #ffd280;
  color: #000000;
}

.darkPrimary {
  background-color: #000000;
  color: #ffffff;
}

.darkSecondary {
  background-color: #404040;
  color: #ffffff;
}

.darkTertiary {
  background-color: #737373;
  color: #ffffff;
}

.disabled {
  background-color: #d9d9d9;
  color: #ffffff;
}
/*# sourceMappingURL=boxStyles.css.map */