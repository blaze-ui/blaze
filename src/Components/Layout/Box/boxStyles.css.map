{
    "version": 3,
    "mappings": "AAEA,AAAA,WAAW,CAAC;EACR,gBAAgB,ECHL,OAAO;EDIlB,KAAK,ECAE,OAAO;CDCjB;;AAED,AAAA,aAAa,CAAC;EACV,gBAAgB,ECPH,OAAO;EDQpB,KAAK,ECLE,OAAO;CDMjB;;AAED,AAAA,YAAY,CAAC;EACT,gBAAgB,ECXJ,OAAO;EDYnB,KAAK,ECVE,OAAO;CDWjB",
    "sources": [
        "boxStyles.scss",
        "../../../Styles/globals.scss"
    ],
    "names": [],
    "file": "boxStyles.css"
}