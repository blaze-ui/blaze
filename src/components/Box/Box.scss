@import "../../styles/ColorSchema.scss";
.Primary {
  background: $primary;
  color: #000000;
}

.Secondary {
  background: linear-gradient(60deg, purple, pink);
  color: #000000;
}

.Tertiary {
  background: linear-gradient(60deg, #ffff00, #ffffff);
  color: #000000;
}

.dark {
  background-color: #000000;
  color: #ffffff;
}

.darkPrimary {
  background: linear-gradient(to left, #ff0000 0%, #000000 80%);
  color: #ffffff;
}

.darkSecondary {
  background: linear-gradient(to left, yellow 0%, #000000 80%);
  color: #ffffff;
}

.disabled {
  background-color: #d9d9d9;
  color: #ffffff;
}
/*# sourceMappingURL=boxStyles.css.map */
