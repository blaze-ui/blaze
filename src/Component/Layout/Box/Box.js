import React,{useState} from 'react';
import './Box.css';

const Box = (props) => {
    // const property = props.property;
    // this color is {props.property}
    return(
        <div className={props.property}>
           "this is awesome"
        </div>
    )
}

export default Box;