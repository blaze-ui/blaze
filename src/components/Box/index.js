/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable linebreak-style */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/button-has-type */
/* eslint-disable react/prefer-stateless-function */
import Proptypes from "prop-types";
import React from "react";
import "./Box.css";
import Typography from "../Typography";

function Box(props) {
  const { color, textColor } = props;
  const boxPadding = `${props.p}em`;
  const boxMargin = `${props.m}em`;

  const _styles = {
    padding: `${boxPadding}`,
    margin: `${boxMargin}`,
    color: `${textColor}`,
    textColor: { textColor }
  };

  return (
    <div className={color} style={_styles}>
      <Typography>
        This is: {color} with textColor: {textColor}
      </Typography>
    </div>
  );
}

Box.defaultProps = {
  color: "black",
  textColor: "white",
  p: "2em",
  m: "2.5em"
};

Box.propTypes = {
  color: Proptypes.string,
  textColor: Proptypes.string,
  p: Proptypes.number,
  m: Proptypes.number
};

export default Box;
