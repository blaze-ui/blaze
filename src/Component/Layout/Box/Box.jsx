import React, { Component } from "react";
import Proptypes from "prop-types";
import "./boxStyles.scss";

export default class Box extends Component {
  render() {
    const boxColor = this.props.color;
    const boxPadding = `${this.props.p}em`;
    const boxMargin = `${this.props.m}em`;

    return (
      <div
        className={boxColor}
        style={{
          padding: `${boxPadding}`,
          margin: `${boxMargin}`
        }}
      >
        This is {this.props.color}
      </div>
    );
  }
}

Box.propType = {
  color: Proptypes.string,
  p: Proptypes.number
};
