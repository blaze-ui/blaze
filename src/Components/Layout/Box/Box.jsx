import React, { Component } from "react";
import Proptypes from "prop-types";
import "./boxStyles.scss";

export default class Box extends Component {
  render() {
    const boxColor = this.props.color;
    const boxPadding = `${this.props.p}em`;
    const boxMargin = `${this.props.m}em`;
    const textColor = this.props.color === "disabled" ? ("black") : ("white");
    return (
      <div
        className={boxColor}
        style={{
          padding: `${boxPadding}`,
          margin: `${boxMargin}`,
          color: `${textColor}`
        }}
      >
        This is {this.props.color}
      </div>
    );
  }
}

Box.propTypes = {
  color: Proptypes.string,
  p: Proptypes.number,
  m: Proptypes.number
};
