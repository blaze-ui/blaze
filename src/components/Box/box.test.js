import React from "react";
import renderer from "react-test-renderer";
import Box from "./index";

test("box default test case", () => {
  const component = renderer.create(<Box color="Primary" p={2} m={2.5} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
