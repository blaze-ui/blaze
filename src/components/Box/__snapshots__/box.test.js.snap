// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`box default test case 1`] = `
<div
  className="Primary"
  style={
    Object {
      "color": "white",
      "margin": "2.5em",
      "padding": "2em",
      "textColor": Object {
        "textColor": "white",
      },
    }
  }
>
  This is: 
  Primary
   with textColor: 
  white
</div>
`;
